services/galley/src/Galley/Effects/BrigAccess.hs

```diff
@@ -128,6 +129,7 @@ data BrigAccess m a where
   GetClientByKeyPackageRef :: KeyPackageRef -> BrigAccess m (Maybe ClientIdentity)
   GetLocalMLSClients :: Local UserId -> SignatureSchemeTag -> BrigAccess m (Set ClientId)
   AddKeyPackageRef :: KeyPackageRef -> Qualified UserId -> ClientId -> Qualified ConvId -> BrigAccess m ()
+  UpdateKeyPackageRef :: KeyPackageRef -> KeyPackageRef -> BrigAccess m ()
```

Comment:
I know we need two key package refs here, but I don't like the fact they're of the same type, which makes it easy to confuse them. Would it make sense or would it be an overkill to introduce a wrapper like this?

```haskell
data Age = New | Old
data Aged t (a :: Age) where
  NewAged :: t -> Aged t 'New
  OldAged :: t -> Aged t 'Old
```

Then this effect action would look like this (and similarly for other places where there are two key package refs in the signature):
```haskell
  UpdateKeyPackageRef :: Aged KeyPackageRef 'Old -> Aged KeyPackageRef 'New -> BrigAccess m ()
```
Use sites would then mark it explicit which one is new, and which one is old:

```haskell
  ...
  updateKeyPackageRef (OldAged prevRef) (NewAged newRef)
```

Reply:
Done, but using a type with two fields. See [12c9dbe](https://github.com/wireapp/wire-server/pull/2604/commits/12c9dbe4b5864d355ad264943a209b212f68adc8).